src/hu_stuff.c

```diff
@@ -1016,30 +1025,31 @@ static void HU_widget_build_weapon (void)
     ammopct = fullammo ? (100 * ammo) / fullammo : 100;
 
     // display each weapon number in a color related to the ammo for it
-    hud_weapstr[i++] = '\x1b'; //jff 3/26/98 use ESC not '\' for paths
+    hud_stringbuffer[i++] = '\x1b'; //jff 3/26/98 use ESC not '\' for paths
     if (weaponinfo[w].ammo == am_noammo) //jff 3/14/98 show berserk on HUD
-      hud_weapstr[i++] = (w == wp_fist && !plr->powers[pw_strength]) ? '0'+CR_GRAY : '0'+CR_GREEN;
+      hud_stringbuffer[i++] = (w == wp_fist && !plr->powers[pw_strength]) ? '0'+CR_GRAY : '0'+CR_GREEN;
     else if (ammopct < ammo_red)
-      hud_weapstr[i++] = '0'+CR_RED;
+      hud_stringbuffer[i++] = '0'+CR_RED;
     else if (ammopct < ammo_yellow)
-      hud_weapstr[i++] = '0'+CR_GOLD;
+      hud_stringbuffer[i++] = '0'+CR_GOLD;
     else if (ammopct > 100) // more than max threshold w/o backpack
-      hud_weapstr[i++] = '0'+CR_BLUE;
+      hud_stringbuffer[i++] = '0'+CR_BLUE;
     else
-      hud_weapstr[i++] = '0'+CR_GREEN;
+      hud_stringbuffer[i++] = '0'+CR_GREEN;
 
-    hud_weapstr[i++] = '0'+w+1;
-    hud_weapstr[i++] = ' ';
-    hud_weapstr[i] = '\0';
+    hud_stringbuffer[i++] = '0'+w+1;
+    hud_stringbuffer[i++] = ' ';
+    hud_stringbuffer[i] = '\0';
   }
 
   // transfer the init string to the widget
-  HUlib_add_string_to_cur_line(&w_weapon, hud_weapstr);
+  HUlib_add_string_to_cur_line(&w_weapon, hud_stringbuffer);
 }
 
 static void HU_widget_build_keys (void)
 {
-  char hud_keysstr[HU_MAXLINELENGTH] = { 'K', 'E', 'Y', '\x1b', '0'+CR_NONE, ' ', '\0' };
+  InitStringBuffer((char[]){ 'K', 'E', 'Y', '\x1b', '0'+CR_NONE, ' ', '\0' });
```

Comment:
I think we've avoided compound literals in the past. Do we want to continue with that philosophy?

Reply:
What do you suggest in this case?

Reply:
Something like this maybe?

```c
const char hud_keysstr[] = { 'K', 'E', 'Y', '\x1b', '0'+CR_NONE, ' ', '\0' };
InitStringBuffer(hud_keysstr);
```

There's another one in `HU_widget_build_frag()` too.